src/backends/sentry_backend_inproc.c

```diff
@@ -87,14 +113,15 @@ startup_inproc_backend(
     }
 
     // install our own signal handler
+#    ifdef SETUP_SIGALTSTACK
     g_signal_stack.ss_sp = sentry_malloc(SIGNAL_STACK_SIZE);
     if (!g_signal_stack.ss_sp) {
         return 1;
     }
     g_signal_stack.ss_size = SIGNAL_STACK_SIZE;
     g_signal_stack.ss_flags = 0;
     sigaltstack(&g_signal_stack, 0);
-
+#    endif
     sigemptyset(&g_sigaction.sa_mask);
     g_sigaction.sa_sigaction = handle_signal;
     g_sigaction.sa_flags = SA_SIGINFO | SA_ONSTACK;
```

Comment:
Just double checking: We still should set the `SA_ONSTACK` flag, even if we don't allocate the altstack ourselves right?

Reply:
Yeah, `SA_ONSTACK` tells the signal handler to use the alt stack or falls back to the process stack. Since the assumption is that every thread has an alt stack, we should be good.